### ISSUE TITLE
kontainer-driver-metadata changes are not updated for rkeaddons

### ISSUE THREAD

**Issue by Oats87** [rancher/rancher#24675]
**What kind of request is this (question/bug/enhancement/feature request):** Bug

**Steps to reproduce (least amount of steps as possible):**

We were trying to modify the `rkeaddon` for `canal-v1.16` (see: https://github.com/HighwayofLife/kontainer-driver-metadata/blob/master/rke/templates/canal.go#L2414)

We generated a new `data.json` and specified this repository in the Settings and triggered off a `rke-metadata-config` refresh.

We could see in the logs that:
```
2019/12/19 01:35:05 [INFO] driverMetadata: refresh data
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.12-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.7-rancher1-3
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.9-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.9.7-rancher2-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.3-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.5-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.9-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.10-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.1-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.1-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.6-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.6-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.6-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.7-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.9-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.6-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.1-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.1-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.8-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.4-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.5-rancher1-3
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.3-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.16.1-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.1-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.3-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.0-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.8.11-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.9.7-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.10-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.7-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.1-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.2-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.16.2-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.6-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.9-rancher1-3
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.9-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.10-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.8-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.4-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.3-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.7-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.12-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.9.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.5-rancher2-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.11-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.2-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.2-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.11-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.5-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.5-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.9-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.7-rancher1-4
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.4-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.9-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.10-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.7-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.16.3-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.17.0-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.0-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.1-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.12.5-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.4-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.0-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.15.3-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.10.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.14.8-rancher2-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.11.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.12-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.4-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.13.5-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8ssystemimages/v1.16.2-rancher1-1
2019/12/19 01:35:05 [DEBUG] driverMetadata deprecated map[] max incompatible versions [v1.10.12-rancher1-1 v1.12.7-rancher1-3 v1.13.9-rancher1-1 v1.9.7-rancher2-2 v1.10.3-rancher2-1 v1.10.5-rancher1-2 v1.13.9-rancher1-2 v1.12.10-rancher1-2 v1.13.1-rancher1-1 v1.15.5-rancher1-1 v1.12.6-rancher1-1 v1.12.6-rancher1-2 v1.12.7-rancher1-1 v1.13.1-rancher1-2 v1.10.1-rancher2-1 v1.11.8-rancher1-1 v1.12.4-rancher1-1 v1.13.5-rancher1-3 v1.11.1-rancher1-1 v1.11.3-rancher1-1 v1.12.0-rancher1-1 v1.8.11-rancher2-1 v1.9.7-rancher2-1 v1.12.5-rancher1-1 v1.13.10-rancher1-2 v1.13.7-rancher1-1 v1.11.6-rancher1-1 v1.11.9-rancher1-3 v1.12.9-rancher1-1 v1.13.10-rancher1-1 v1.12.3-rancher1-1 v1.12.7-rancher1-2 v1.13.12-rancher2-1 v1.9.5-rancher1-1 v1.10.11-rancher1-1 v1.11.2-rancher1-1 v1.11.2-rancher1-2 v1.13.11-rancher1-1 v1.13.5-rancher1-2 v1.11.9-rancher1-2 v1.12.7-rancher1-4 v1.11.9-rancher1-1 v1.12.10-rancher1-1 v1.10.0-rancher1-1 v1.12.1-rancher1-1 v1.12.5-rancher1-2 v1.13.4-rancher1-1 v1.10.5-rancher1-1 v1.11.5-rancher1-1 v1.13.12-rancher1-1 v1.13.4-rancher1-2 v1.13.5-rancher1-1]
2019/12/19 01:35:05 [DEBUG] driverMetadata: userSettings map[]
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8sserviceoptions/v1.16.3-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8sserviceoptions/v1.15.6-rancher1-2
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8sserviceoptions/v1.14.9-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkek8sserviceoptions/v1.17.0-rancher1-1
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkeaddons/canal-v1.16
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkeaddons/nginxingress-v1.15
2019/12/19 01:35:06 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkeaddons/canal-v1.15
2019/12/19 01:35:06 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkeaddons/coredns-v1.17
2019/12/19 01:35:06 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkeaddons/nginxingress-v1.8
```

namely the 
```
2019/12/19 01:35:05 [DEBUG] REST UPDATE apis/management.cattle.io/v3/cattle-global-data/rkeaddons/canal-v1.16
``` 
line says this should have been updated

**Result:**
When I check the `rkeaddon` for `canal-v1.16` I don't see my changes. The only way I got my changes to show up was to delete the `canal-v1.16` `rkeaddon` and then trigger off an update.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3.3
- Installation option (single install/HA): HA

**Comment by jiaqiluo** [comment]
The bug is reproduced in Rancher v2.3.3

Steps:
- clone the kontainer-driver-metadata repo: https://github.com/rancher/kontainer-driver-metadata
- create a new branch for testing
- edit the settings in Rancher to points to the new branch in that repo
- in Rancher under `v3/rkeaddons` check the template for Canal 1.16 to  see the following line containing the value: ` path: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds`
```
# Used to install Flex Volume Driver\n        - name: flexvol-driver-host\n          hostPath:\n            type: DirectoryOrCreate\n            path: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds\n
```
- change the path to `/foo/bar` in the branch in the cloned repo, and generate and push new data file
- refresh kubernetes metadata in Rancher (Tools -> Driver)
- refresh the  `v3/rkeaddons` page

Result:
- the data is NOT updated to reflect the change
<img width="995" alt="Screen Shot 2019-12-30 at 2 07 26 PM" src="https://user-images.githubusercontent.com/6218999/71600776-ba656d80-2b0d-11ea-84ea-da47489d9239.png">

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: v2.3-head

Steps:
the same as the previous comment provides

Result:
- the data is updated to reflect the change 
<img width="1007" alt="Screen Shot 2019-12-30 at 2 21 12 PM" src="https://user-images.githubusercontent.com/6218999/71601188-a7ec3380-2b0f-11ea-9c0a-430c117e4410.png">

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: master-head f875cf6a2

Steps:
the same as the previous comment provides

Result:
- the data is updated to reflect the change 

<img width="1080" alt="Screen Shot 2019-12-30 at 2 22 18 PM" src="https://user-images.githubusercontent.com/6218999/71601251-d1a55a80-2b0f-11ea-8ff6-48c98e9a8d4c.png">
